packedForest/src/forestTypes/unsupervisedForests/urerf/urerfTree.h

```diff
@@ -0,0 +1,267 @@
+                                                                                        pairMat.insert({pair1, 1});
+                                                                                }
+                                                                        }
+                                                                }
+                                                        }
+                                                }
+                                        }
+
+                                inline void updateSimMatOut(std::map<int, std::map<int, int> > &simMat, std::map<std::pair<int, int>, double> &pairMat){
+                                        for(auto nodes : leafNodes){
+                                                stratifiedInNodeClassIndicesUnsupervised* obsI = nodes.returnObsIndices();
+                                                std::vector<int> leafObs;
+
+                                                leafObs = obsI->returnOutSampsVec();
+                                                auto siz = leafObs.size();
+                                                if (siz <= 0)
+                                                        continue;
+
+                                                for(unsigned int i = 0; i < siz; ++i) {
+                                                        for (unsigned int j=0; j<=i; ++j) {
+                                                                std::pair<int, int> pair1 = std::make_pair(leafObs[i], leafObs[j]);
+                                                                #pragma omp critical
+                                                                {
+                                                                auto it = pairMat.find(pair1);
+                                                                if(it!=pairMat.end())
+                                                                        pairMat[pair1]++;
+                                                                else
+                                                                        pairMat.insert({pair1, 1});
+                                                                }
+                                                        }
+                                                }
+                                        }
+                                }
+
+				inline int returnLeafDepthSum(){
+					int leafDepthSums=0;
+					for(auto nodes : tree){
+						if(!nodes.isInternalNode()){
+							leafDepthSums += nodes.returnDepth();
+						}
+					}
+					return leafDepthSums;
+				}
+
+
+				inline void setAsLeaf(){
+					tree.back().setClass(nodeQueue.back().returnMaxClass());
+					tree.back().setDepth(nodeQueue.back().returnDepth());
+				}
+
+
+				inline void checkOOB(){
+					totalOOB += nodeQueue.back().returnOutSampleSize();
+					correctOOB += nodeQueue.back().returnOutSampleError(tree.back().returnClass());
```

Comment:
I think this OOB stuff needs updated. remove the `correctOOB ...` line
cf. packedForest/src/forestTypes/basicForests/ferf/rerfTree.h

```
					// Get oob_indices and vote from leaf node as
					// <(oob_id, vote)> and push back onto the
					// indexAndVote object.
					for(auto &j : nodeQueue.back().returnOutSampleIdsFromLeaf()){
						indexAndVote.push_back(std::vector<int>{j, tree.back().returnClass()});
					}
```

Reply:
checkOOB is not applicable for unsupervised random forests so I added an error message.